sql_builder.go

```diff
@@ -83,7 +82,12 @@ func (sq *sqlBuilder) compile() {
 				}
 				sq.sql = sq.Query.RawSQL.Fragment
 			}
+			sq.args = sq.Query.RawSQL.Arguments
 		} else {
+			if sq.Model == nil {
+				sq.err = fmt.Errorf("sqlBuilder.compile() called but no RawSQL and Model specified")
+				return
```

Comment:
1. prevent execution of the following calls that require Model or RawQuery
2. set error message to the caller can see it

now the caller will not call the builder, but to make it safer.